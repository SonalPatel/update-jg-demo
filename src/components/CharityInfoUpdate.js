import React from "react";
import { getCharityById } from "./../helper/Api";

class CharityInfoUpdate extends React.Component {
  //settings and state
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      charityContent: [] // my empty array
    };
  }

  //when the component has mounted (page has loaded) we want to perform our ajax call using Fetch which is the modern way to make an ajax call
  //fetch returns a promise which contains various information.
  // the headers were required as the api was xml: if its not json then make it json.
  componentDidMount() {
    // call my re-useable function and manually enter the charity ID  (cruk: 2357, oxfam:13441, bhf: 183092 ).
    //getCharityById(2357)
    getCharityById(this.props.charityId)
      // taking  the previous promise, reading it and using another promise to make into Json (res) (.then is like a callback function)
      .then(res => res.json())
      //update the state with the info from the api (state is like a holding place, setState is a react method)
      .then(
        result => {
          this.setState({
            isLoaded: true,
            charityContent: result
          });
        },
        //error function just in case (error is a built in react property)
        error => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      );
  }

  render() {
    //console.log(this.props.charityId);
    const { error, isLoaded, charityContent } = this.state;
    //check if error then display an error message {error.message} is part of react
    if (error) {
      return <div>Error: {error.message} </div>;
    }

    //if the content has not loaded yet, then show the text loading message
    //if (!isLoaded) {
    if (isLoaded === false) {
      return (
        <div className="brand-primary">
          <p>Loading charity content...</p>
          <div className="lds-circle" />
        </div>
      );
    }

    // returns the content from the api and then iterates over each object to allow me access to the info
    //(charityContent is my varible name for the instance of each index defined in the settings at the top);
    return (
      <div className="content-info" id="content" role="main">
        <div className="charity-wrapper">
          <div className="charity-logo-wrap">
            <img
              className="charity-logo img-fluid"
              src={charityContent.logoAbsoluteUrl}
              alt={`${charityContent.name} Logo`}
            />
          </div>

          <h1 className="off-screen">{charityContent.name}</h1>
        </div>
        <div className="charity-info">
          <h2>
            {charityContent.name} {charityContent.impactStatementWhat} to{" "}
            {charityContent.impactStatementWhy}
          </h2>

          <p>{charityContent.description}</p>
          <a href={charityContent.profilePageUrl}>
            Find out more about {charityContent.name}{" "}
          </a>
        </div>

        <aside role="complementary">
          <ul className="list-items">
            <li>
              <strong>Contact us at: </strong>
              <a href={`mailto:${charityContent.emailAddress}`}>
                {charityContent.emailAddress}
              </a>
            </li>
            <li>
              <strong>{charityContent.name} registered charity number:</strong>{" "}
              {charityContent.id}
            </li>
          </ul>
        </aside>
      </div>
    );
  }
}

export default CharityInfoUpdate;
