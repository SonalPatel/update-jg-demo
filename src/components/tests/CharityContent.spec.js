import React from "react";
import { shallow } from "enzyme";
import CharityInfoUpdate from "../CharityInfoUpdate";
import formatted from "enzyme-to-json"; //prettify is my variable name i can call it anything

//take a snapshot of the loading state in the CHARITY component **********CHECK***************
describe("Render CharityUpdate component", () => {
  // when the state is set to loading, take a snapshot of the html code in a readable format
  it("should render in a loading state", () => {
    const charityInfoUpdate = shallow(<CharityInfoUpdate />);
    //formatt the code so it's easy to read
    const formatted = prettify(charityInfoUpdate);
    expect(formatted).toMatchSnapshot();
  });

  //take a snapshot of the whole CharityUpdate component rendered html when it has sucsessfully loaded with no errors
  it("should render full component", () => {
    const state = {
      error: null,
      isLoaded: true,
      charityContent: []
    };
    const charityInfoUpdate = shallow(<CharityInfoUpdate />);
    charityInfoUpdate.setState(state);

    //formatt the code so it's easy to read
    const formatted = prettify(charityInfoUpdate);
    expect(formatted).toMatchSnapshot();
  });
});
