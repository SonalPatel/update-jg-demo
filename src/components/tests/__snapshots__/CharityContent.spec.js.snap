// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CharityUpdate component should render full component 1`] = `
<div
  className="content-info"
  id="content"
  role="main"
>
  <div
    className="charity-wrapper"
  >
    <div
      className="charity-logo-wrap"
    >
      <img
        alt="undefined Logo"
        className="charity-logo img-fluid"
      />
    </div>
    <h1
      className="off-screen"
    />
  </div>
  <div
    className="charity-info"
  >
    <h2>
       
       to
       
    </h2>
    <p />
    <a>
      Find out more about 
       
    </a>
  </div>
  <aside
    role="complementary"
  >
    <ul
      className="list-items"
    >
      <li>
        <strong>
          Contact us at: 
        </strong>
        <a
          href="mailto:undefined"
        />
      </li>
      <li>
        <strong>
           registered charity number:
        </strong>
         
      </li>
    </ul>
  </aside>
</div>
`;

exports[`Render CharityUpdate component should render in a loading state 1`] = `
<div
  className="brand-primary"
>
  <p>
    Loading charity content...
  </p>
  <div
    className="lds-circle"
  />
</div>
`;
